File: src/main/java/bad/robot/radiate/NullProgress.java
package bad.robot.radiate;

import static org.apache.commons.lang3.builder.EqualsBuilder.reflectionEquals;
import static org.apache.commons.lang3.builder.HashCodeBuilder.reflectionHashCode;

@Deprecated
public class NullProgress extends Progress {

    public NullProgress() {
        super(0, 0);
    }

    @Override
    public void increment() { }

    @Override
    public void decrement() { }

    @Override
    public Progress add(Progress progress) {
        return super.add(progress);
    }

    @Override
    public int asAngle() {
        return 0;
    }

    @Override
    public boolean complete() {
        throw new UnsupportedOperationException();
    }

    @Override
    public boolean lessThan(Progress progress) {
        throw new UnsupportedOperationException();
    }

    @Override
    public boolean greaterThan(Progress progress) {
        throw new UnsupportedOperationException();
    }

    @Override
    public int numberOfBuilds() {
        return 0;
    }

    @Override
    public String toString() {
        return "No progress";
    }

    @Override
    public boolean equals(Object that) {
        return reflectionEquals(this, that);
    }

    @Override
    public int hashCode() {
        return reflectionHashCode(this);
    }
}

File: src/main/scala/bad/robot/radiate/NullProgress.scala
package bad.robot.radiate

class NullProgressS extends ProgressS(0, 0) {

  override def increment() {}

  override def decrement() {}

  override def add(progress: ProgressS) = super.add(progress)

  override def asAngle = 0

  override def complete = throw new UnsupportedOperationException

  override def <(progress: ProgressS) = throw new UnsupportedOperationException

  override def >(progress: ProgressS) = throw new UnsupportedOperationException

  override def numberOfBuilds = 0

  override def toString = "No progress"

}